### PR TITLE
Support not equal operations in sql conditions

### PR DESCRIPTION
**Pull request by crabtree**
**Description**

Changes proposed in this pull request:

- Support not equal operations in sql conditions

**Related issue(s)**

See also #366

### CODE REVIEW

**Comment by crabtree** [comment]
@dbadura I agree it is worth covering with e2e, in case of unit tests it is already covered in the repository for system auths.

File: components/director/internal/repo/common.go [hunk 1]
```diff
@@ -6,18 +6,65 @@ import (
...
 
+func NewEqualCondition(field, val string) Condition {
+	return Condition{
+		Field: field,
+		Val:   val,
+		Op:    EqualOp,
+	}
+}
+
+func NewNotEqualCondition(field, val string) Condition {
```
Comment (aszecowka) [inline components/director/internal/repo/common.go]: this method seems to not be used, even in tests: `NewNotEqualCondition` and `NewNullCondition`

File: components/director/internal/repo/common.go [hunk 2]
```diff
@@ -6,18 +6,65 @@ import (
 	"strings"
 )
 
+type ConditionOp string
+
+const (
+	EqualOp    ConditionOp = "="
```
[the remark this answers is not in the document]
Comment (crabtree) [reply components/director/internal/repo/common.go]: Had some discussion previously in a different case with @pkosiec and he suggested to be more descriptive when naming the consts.

File: components/director/internal/repo/common.go (hunk 1 above)
Comment (crabtree) [reply components/director/internal/repo/common.go]: removed

File: components/director/internal/domain/systemauth/service.go [hunk 3]
```diff
@@ -15,6 +15,7 @@ type Repository interface {
 	GetByID(ctx context.Context, tenant, id string) (*model.SystemAuth, error)
 	ListForObject(ctx context.Context, tenant string, objectType model.SystemAuthReferenceObjectType, objectID string) ([]model.SystemAuth, error)
 	Delete(ctx context.Context, tenant string, id string) error
+	DeleteByIDForObject(ctx context.Context, tenant string, id string, objType model.SystemAuthReferenceObjectType) error
```
[the remark this answers is not in the document]
Comment (crabtree) [reply components/director/internal/domain/systemauth/service.go]: The name is just to be consistent with the `ListForObject`

[the remark this answers is not in the document]
Comment (crabtree) [reply components/director/internal/domain/systemauth/service.go]: nope, removed

File: tests/end-to-end/director/oauth20_test.go [hunk 4]
```diff
@@ -135,6 +137,56 @@ func TestDeleteSystemAuthFromApplication(t *testing.T) {
 	require.Empty(t, appAfterDelete.Auths)
 }
 
+func TestDeleteSystemAuthFromApplicationUsingRuntimeMutation(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (crabtree) [reply tests/end-to-end/director/oauth20_test.go]: done

File: tests/end-to-end/director/oauth20_test.go [hunk 5]
```diff
@@ -190,3 +292,83 @@ func TestDeleteSystemAuthFromIntegrationSystem(t *testing.T) {
...
+
+	// WHEN
+	t.Log("Delete system auth for integration system using runtime mutation")
+	err := tc.RunOperation(ctx, deleteSystemAuthForRuntimeRequest, &deleteOutput)
+
+	//THEN
+	require.Error(t, err)
+}
+
+func deleteClientCredentialsForApplication(t *testing.T, authID string) {
```
Comment (dbadura) [inline tests/end-to-end/director/oauth20_test.go]: What do you think about moving all those  to the `fixtures_queries_test.go`?

Comment (crabtree) [reply tests/end-to-end/director/oauth20_test.go]: If this is the convention, previously I saw that such methods are defined within its domain, which I prefer.

**Pull request merged**